### PR TITLE
backup_and_restore.md

### PR DESCRIPTION
**Pull request by dmitrypekar**
Please review & merge a doc for backup and restore procedures.
Backup implementation is already merged while restore is in progress (PR created).

### CODE REVIEW

File: docs/docs/backup-and-restore.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
+---
+title: Backup and Restore
+---
+
+# Backup and Restore
+
+Cassandra on Mesos supports backup and restore operation. The API for those operation is exposed through REST.
```
Comment (ssk2) [inline docs/docs/backup-and-restore.md]: `Cassandra on Mesos` -> `Cassandra-Mesos`

Comment (ssk2) [inline docs/docs/backup-and-restore.md]: `The API for those operation is exposed through REST` -> `These operations are exposed through a REST API.`

File: docs/docs/backup-and-restore.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+# Configuration
+
+Configuration is provided via environment variable `CASSANDRA_BACKUP_DIRECTORY`. This directory specifies there
```
Comment (ssk2) [inline docs/docs/backup-and-restore.md]: `via environment variable` -> `via the environment variable`.

Comment (ssk2) [inline docs/docs/backup-and-restore.md]: `there` -> `where`

File: docs/docs/backup-and-restore.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+to store named backups. Default values is `backup` directory inside slave's sandbox.
```
Comment (ssk2) [inline docs/docs/backup-and-restore.md]: `Default values is `backup` directory inside slave's sandbox.` -> `The default directory is `backup` inside the task's sandbox.`

File: docs/docs/backup-and-restore.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+# Example
+
+Suppose we have started Cassandra on Mesos cluster having 2+ nodes.
```
Comment (ssk2) [inline docs/docs/backup-and-restore.md]: `Cassandra on Mesos` -> `Cassandra-Mesos`

File: docs/docs/backup-and-restore.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+3. Data is reloaded (using loadNewSSTables JMX call);
+
+# Limitations
+
+## No atomicity.
+Backup and Restore procedures are not atomic. If DB is actively modified during backup or restore both procedures could
+produce inconsistent (from the user's perspective) data sets. So it is recommended to create a backup (and restore from
+a backup) during DB inactivity time.
+
+## Executor id preservation
```
Comment (ssk2) [inline docs/docs/backup-and-restore.md]: `id` -> `ID`

File: docs/docs/backup-and-restore.md (hunk 1 above)
Comment (dmitrypekar) [inline docs/docs/backup-and-restore.md]: Fixed both.

File: docs/docs/backup-and-restore.md (hunk 2 above)
Comment (dmitrypekar) [inline docs/docs/backup-and-restore.md]: Fixed both.

File: docs/docs/backup-and-restore.md (hunk 3 above)
Comment (dmitrypekar) [inline docs/docs/backup-and-restore.md]: Fixed.

File: docs/docs/backup-and-restore.md (hunk 4 above)
Comment (dmitrypekar) [inline docs/docs/backup-and-restore.md]: Fixed.

File: docs/docs/backup-and-restore.md (hunk 5 above)
Comment (dmitrypekar) [inline docs/docs/backup-and-restore.md]: Fixed.

**Comment by elingg** [comment]
hi @dmitrypekar, could you please describe here how to determine if there was an error in any keyspace and also let the user know that they need to restart the job from the beginning?

**Comment by elingg** [comment]
@MattPennaThe3rd, could you please take a second look, thanks! I also left one comment above that needs to be addressed.

**Comment by dmitrypekar** [comment]
Hi @elingg, @mattpennathe3rd 

I've made a following changes:
1. pushed this backup-and-restore.md to the restore_basic2 PR (this is required because this branch became orphan after I've reforked project 2 weeks ago).
2. added a section in restore_basic2 PR about handling errors, that describe how to interpret the details of the backup|restore operations and what to do.

Please do following:
1. close this PR as expired;
2. continue review of this of document using restore_basic2 PR;
3. merge it as a part of restore_basic PR if no problem needs to be addressed;

**Comment by elingg** [comment]
Ok, we will close this and give any additional feedback on restore_basic2 PR

**Pull request closed without merging**